test/tools/utils.ts

```diff
@@ -499,3 +504,58 @@ export function topologyWithPlaceholderClient(
     options as TopologyOptions
   );
 }
+
+export async function itInNodeProcess(
+  title: string,
+  fn: (d: { expect: typeof import('chai').expect; mongodb: typeof import('../mongodb') }) => void
+) {
+  it(title, async () => {
+    const script = `
+      import { expect } from 'chai';
+      import * as mongodb from './test/mongodb';
+      const run = ${fn};
+      run({ expect, mongodb }).then(
+        () => process.exit(0),
+        error => {
+          console.error(error)
+          process.exit(1)
+        }
+      );\n`;
+
+    const scriptName = `./testing_${title.split(/\s/).join('_')}_script.cts`;
+    const cwd = path.resolve(__dirname, '..', '..');
+    const tsNode = path.resolve(__dirname, '..', '..', 'node_modules', '.bin', 'ts-node');
+    try {
+      await fs.writeFile(scriptName, script, { encoding: 'utf8' });
+      const scriptInstance = child_process.fork(scriptName, {
+        signal: AbortSignal.timeout(50_000),
+        cwd,
+        stdio: 'pipe',
+        execArgv: [tsNode]
+      });
+
+      scriptInstance.stdout?.setEncoding('utf8');
+      scriptInstance.stderr?.setEncoding('utf8');
+
+      let stdout = '';
+      scriptInstance.stdout?.addListener('data', data => {
+        stdout += data;
+      });
+
+      let stderr = '';
+      scriptInstance.stderr?.addListener('data', (data: string) => {
+        stderr += data
+          .split('\n')
+          .filter(line => !line.startsWith('Debugger') && !line.startsWith('For help'))
+          .join('\n');
```

Comment:
Also just a very minor note, but `data` chunks could also come in as `Debu`, `gger\n` in two separate events (so it might be preferable to do this filtering after all data has been read)

Reply:
Good call